File: 1282.用户分组.go
/*
 * @lc app=leetcode.cn id=1282 lang=golang
 *
 * [1282] 用户分组
 */

// @lc code=start
func groupThePeople(groupSizes []int) [][]int {
	n := len(groupSizes)
	groups := make(map[int][]int)
	for i, v := range groupSizes {
		groups[v] = append(groups[v], i)
	}
	ans := make([][]int, 0)
	for k, v := range groups {
		for i := 0; i < len(v); i += k {
			ans = append(ans, v[i:i+k])
		}
	}
	return ans

}

// @lc code=end

File: 1282.用户分组.py
#
# @lc app=leetcode.cn id=1282 lang=python3
#
# [1282] 用户分组
#

# @lc code=start
class Solution:
    def groupThePeople(self, groupSizes: List[int]) -> List[List[int]]:
        d = defaultdict(list)
        for idx, val in enumerate(groupSizes):
            d[val].append(idx)
        res = []
        for k, v in d.items():
            for i in range(0, len(v), k):
                res.append(v[i: i + k])
        return res
# @lc code=end

